### PR TITLE
Fail Deployment playbook during VIM strategy

### PR DESCRIPTION
**Pull request by enzocandotti**
This commit aims to modify the playbook to fail if there is an ongoing VIM strategy and we are in the midst of a Day-2 operation.

To achieve this, prior to applying the deployment configuration, we need to check for an alarm related to VIM strategies (those with IDs starting with 900) and also verify if the deploymentScope is set as "principal" in the deploy-config file.

Test Plan:
PASS: Create a dummy 900.xxx alarm and set the Scope as "principal" in the deployment config file. Verify that the playbook fails and the correct message is displayed.
PASS: Test both cases where either one or none of the conditions are present (a 900 alarm created or the Scope set as "principal"). Verify that the playbook is executed correctly.

### CODE REVIEW

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml [hunk 1]
```diff
@@ -312,6 +312,25 @@
             timeout: 10
             msg: Waiting for the Deployment Manager validation webhooks to start
 
+        - name: Search for 'principal' scope in deploy_config
+          shell: "grep -E '^[[:blank:]]+deploymentScope:[[:blank:]]+\"principal\"' {{ deploy_config }}"
```
[the remark this answers is not in the document]
Comment (yjian118) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:316]: `1 deploymentScope: principal
 2 deploymentScope:
        "principal"`

Use code snippet to correct the format above

**Comment by enzocandotti** [comment]
The following scenarios have been tested:
Scope NOT set as "principal". Expected that the playbook doesn't fail.

` deploymentScope:
    # "principal"
    "bootstrap"`

 `# deploymentScope: "principal"

  deploymentScope: "bootstrap"

  deploymentScope: # principal
    "bootstrap"

  deploymentScope: 
    principal

   deploymentScope: principal`

  
Scope set as "principal". When a 900 alarm is present, the playbook fails.
`
  deploymentScope:
      "principal"

  deploymentScope: "principal"

  deploymentScope: # bootstrap
    "principal"

  deploymentScope:
    # "bootstrap"
    "principal"

  deploymentScope: # comment
    # bootstrap
    "principal"

  deploymentScope:
    #comment1
    #comment2 
    "principal"`

The regex: '\n\s*deploymentScope:\s*\n*(?:(?:\s*#.*)?\n)*\s*"principal"\s*
    \n: Matches a newline character.
    \s*: Matches zero or more whitespace characters.
    deploymentScope:: Matches the literal text "deploymentScope:".
    \s*: Matches zero or more whitespace characters.
    \n*: Matches zero or more newline characters.
    (?:...): Non-capturing group.
    (?:\s*#.*)?: Optionally matches zero or more whitespace characters followed by a # character and any characters after it until the end of the line.
    \n: Matches a newline character.
    *: Matches zero or more occurrences of the preceding group (?:\s*#.*)?\n.
    \s*: Matches zero or more whitespace characters.
    "principal": Matches the literal text "principal" enclosed in double quotes.
    \s*: Matches zero or more whitespace characters.

In summary, this regular expression is designed to match the following pattern:

  -  A line that starts with "deploymentScope:" followed by optional whitespace characters.
  -  Zero or more lines that may start with whitespace and a # character, with any characters after it until the end of the line.
  -  The final line should contain the exact string "principal" enclosed in double quotes, with optional whitespace characters before and after it.

This pattern allows for flexibility in terms of comments (lines starting with #) between the lines containing "deploymentScope:" and "principal".

**Review by ttakenak — APPROVED** [review]

**Review by sselvara1 — COMMENTED** [review]
LGTM. Noted comment from Yuxing regarding regex is addressed.

**Review by yjian118 — APPROVED** [review]
